python/src/llvm.cc

```diff
@@ -261,20 +261,27 @@ void init_triton_llvm(py::module &&m) {
     });
   });
 
-  m.def("link_extern_lib", [](llvm::Module *mod, std::string path) {
-    llvm::SMDiagnostic err;
-    auto &ctx = mod->getContext();
-    auto extMod = llvm::parseIRFile(path, err, ctx);
-    if (!extMod) {
-      llvm::errs() << "Failed to load " << path;
-      return;
-    }
-    extMod->setTargetTriple(mod->getTargetTriple());
-    extMod->setDataLayout(mod->getDataLayout());
-    if (llvm::Linker::linkModules(*mod, std::move(extMod),
-                                  llvm::Linker::Flags::LinkOnlyNeeded)) {
-      llvm::errs() << "Failed to link " << path;
+  m.def("link_extern_libs", [](llvm::Module *dstMod,
+                               const std::vector<std::string> &paths) {
+    if (paths.empty())
       return;
+
+    LLVMContext &ctx = dstMod->getContext();
+    llvm::Linker linker(*dstMod);
+    for (const std::string &path : paths) {
+      llvm::SMDiagnostic err;
+      std::unique_ptr<llvm::Module> libMod = llvm::parseIRFile(path, err, ctx);
+      if (!libMod) {
+        llvm::errs() << "Failed to load " << path;
```

Comment:
You're not changing it in this PR, but I wonder if you want to raise a Python exception in this case and the other error case below?

Reply:
Yup! I'm a bit scared about exceptions, but hey this is Python land. :D